Src/AutoNSubstituteUnitTest/Scenario.cs

```diff
@@ -0,0 +1,97 @@
+﻿using NSubstitute;
+using NSubstitute.Exceptions;
+using Ploeh.AutoFixture.Xunit;
+using Xunit;
+using Xunit.Extensions;
+
+namespace Ploeh.AutoFixture.AutoNSubstitute.UnitTest
+{
+    public class Scenario
+    {
+        [Theory, AutoDataWithAutoSubstituteCustomization]
+        public void InterfacesAreSubstitutedByDefault(IInterface i, object obj)
+        {
+            i.Method(obj);
+            i.Received().Method(obj);
+        }
+
+        [Theory, AutoDataWithAutoSubstituteCustomization]
+        public void AbstractTypesAreSubstitutedByDefault(AbstractType t, object obj)
+        {
+            t.AbstractMethod(obj);
+            t.Received().AbstractMethod(obj);
+        }
+
+        [Theory, AutoDataWithAutoSubstituteCustomization]
+        public void ConcreteTypesAreNotSubstitutedByDefault(ConcreteType t, object obj)
+        {
+            t.VirtualMethod(obj);
+            Assert.Throws<NotASubstituteException>(() => t.Received().VirtualMethod(obj));
+        }
+
+        [Theory, AutoDataWithAutoSubstituteCustomization]
+        public void ConcreteTypesAreSubstitutedWhenParameterIsMarkedWithSubstituteAttribute(
+            [Substitute]ConcreteType t, object obj)

```

Comment:
It's a fair point about the reason for choosing NSubstitute in the first place; I accept that as a motivating factor for designing this particular feature as an attribute, although I consider the overall design of the NSubstitute API fundamentally flawed.

Anyway, with this overall design question resolved, I will now do a more detailed review of this pull request.